nck/readers/mytarget_reader.py

```diff
@@ -0,0 +1,408 @@
+
+        Args:
+            name_content (str): string representing key of parameters config dict
+            offset (int, optional): potential offset of the request. Defaults to 0.
+            campaign_id (int, optional): potential campaign id of the request. Defaults to 0.
+            banner_ids (list, optional): potential banner ids list of the request. Defaults to [].
+            specific_ban_id (int, optional): id to parse to the url. Defaults to 0.
+
+        Returns:
+            Dict[str, Any]: dict resulting from the request we made to the api
+        """
+        parameters = self.__generate_params_dict(name_content, offset=offset, campaign_id=campaign_id, banner_ids=banner_ids)
+        request = self.__create_request(name_content, parameters, specific_ban_id=specific_ban_id)
+        return requests.get(**request).json()
+
+    def __create_request(
+        self,
+        name_content: str,
+        parameters: Dict[str, Any],
+        specific_ban_id=0
+    ) -> Dict[str, Any]:
+        """This function creates the dict with all the parameters required to query the api
+
+        Args:
+            name_content (str): string representing key of parameters config dict
+            parameters (Dict[str, Any]): dict of parameters retrieved from get_params_dict
+            specific_ban_id (int, optional): id to build the url if required. Defaults to 0.
+
+        Returns:
+            Dict[str, Any]: dict used to make a request to the api
+        """
+        req_base = {
+            'url': self.__get_url(name_content, specific_ban_id=specific_ban_id),
+            'headers': self.__get_header(REQUEST_CONFIG[name_content]['headers_type'])
+        }
+        return {**req_base, **parameters}
+
+    def __get_url(
+        self,
+        name_content: str,
+        specific_ban_id=0
+    ) -> str:
+        """This function retrieves the url and if it is mandatory to add an id in the url
+        we fill if using substitute.
+
+        Args:
+            name_content (str): string representing key of parameters config dict
+            specific_ban_id (int, optional): id to parse to the url. Defaults to 0.
+
+        Returns:
+            str: url endpoint
+        """
+        url = REQUEST_CONFIG[name_content]['url']
+        if name_content == 'get_banner_names':
+            url = url.substitute(id=specific_ban_id)
+        return url
+
+    def __get_header(self, header_type: str):
+        if header_type == "content_type":
+            return {
+                'Content-Type': 'application/x-www-form-urlencoded',
+                'Host': 'target.my.com'
+            }
+        elif header_type == "authorization":
+            return {
+                'Authorization': 'Bearer ' + self.agency_client_token['access_token'],
+                'Host': 'target.my.com'
+            }
+        else:
+            logging.error("No such kind of header available")
```

Comment:
We control the header type, aren't we? So, will this case occur? If no, we can simplify and remove this else block. Otherwise, it would be great to log the `header_type`